### DISCUSSION TITLE
Would Magma allow me to put 4G devices and a Linux server on the same controlled LAN subnet (or two connected subnets)?

### DISCUSSION THREAD

**Discussion by philCryoport** [magma/magma#15415] [Q&A]
Hi folks, I haven't learned the jargon yet, so I'm going to ask a very high-level question that might be best suited for the FAQ for DIY-cellular-network n00bs like myself:

We are testing a set of 4G devices that talk to a cloud server.

Rather than add the variables of 
(a) a public cellular carrier, 
(b) our company's cellular connection,
(c) the routing of the public Internet, and
(d) our cloud server provider

...it would be great if we could remove those variables for the initial testing -- and instead either:

1) put the 4G devices and a test server on the same controlled LAN subnet
-- or --
2) put the 4G devices on a private LAN subnet 1, put the test server on a private LAN subnet 2, and the connect the two subnets with a physical or virtual Layer 3 gateway / router

...because if the devices and test server can't communicate in that basic environment, how are they going to handle communicating i the real world?

When we've proven that the devices and test server can communicate, we'll then add back in the above variables in a controlled manner to make sure that -- when uncontrolled variables don't work perfectly -- the devices and server are resilient.

Does Magma allow me to do this?  If so, which hardware & software should I be looking at.  We're talking a little TINY setup.

Please note that I searched for "LAN" and for "subnet" before this -- and none of the discussions seemed to answer the above question.  I've also tried to RTFM the Magma docs...but without an understanding of the jargon, I'm kinda lost...

Thank you for your assistance!

**Comment by juliogaraujo** [comment]
Hello,

I'm not sure if I understood 100% of your needs, but I think you are looking for the AGW Non-NAT Mode configuration so I would suggest you to take a look in the following how-to documentation: https://magma.github.io/magma/docs/howtos/config_agw_bridged.

I already used this feature with DHCP and I was able to manage the UE to get an IP address from my external DHCP Server where I could have total control about the network that UE is using in my network.

Hope this can help you.

**Reply by philCryoport** [reply]
Thank you @juliogaraujo give me a week or two to digest the document and to verify that my team is happy with it...

**Reply by jblakley** [reply]
Didn't know about AGW non-NAT mode. That's an interesting and useful capability! Thanks.

In our private network, we use AGW in NAT Mode. The UE gets an IP from the AGW and the AGW SGi interface is connected to our private  LAN. This seems pretty similar to your #2 model with the AGW acting as the gateway between the two subnets.
